File: Java/100_remove_duplicates_from_sorted_array.java
public class Solution {
    /*
     * @param nums: An ineger array
     * @return: An integer
     */
    public int removeDuplicates(int[] nums) {
        if(nums == null || nums.length == 0) {
            return 0;
        }
        int slow = 0;
        int fast = 1;
        while(fast < nums.length) {
            if(nums[slow] != nums[fast]) {
                nums[++slow] = nums[fast];
            }
            ++fast;
        }
        return slow + 1;
    }
}

File: cpp/100_remove_duplicates_from_sorted_array.cpp
// Time:  O(n)
// Space: O(1)

class Solution {
public:
    /**
     * @param A: a list of integers
     * @return : return an integer
     */
    int removeDuplicates(vector<int> &nums) {
        if (nums.empty()) {
            return 0;
        }
        auto slow = nums.begin();
        auto fast = slow + 1;
        while (fast != nums.end()) {
            if (*fast != *slow) {
                *(++slow) = *fast;
            }
            ++fast;
        }
        nums.resize(slow - nums.begin() + 1);
        return nums.size();
    }
};
